src/monattack.cpp

```diff
@@ -2982,6 +2985,62 @@ bool mattack::nurse_operate( monster *z )
     z->anger = 0;
     return false;
 }
+bool mattack::check_money_left( monster *z )
+{
+    if( !z->has_effect( effect_pet ) ) {
+        if( z->friendly == -1 &&
+            z->has_effect( effect_paid ) ) { // if the pet effect runs out we're no longer friends
+            z->friendly = 0;
+            bool had_inventory = !z->inv.empty();
+            for( const item &it : z->inv ) {
+                g->m.add_item_or_charges( z->pos(), it );
+            }
+            z->inv.clear();
+            z->remove_effect( effect_has_bag );
+            if( had_inventory ) {
+                add_msg( m_info,
+                         _( "The %s dumps the contents of its bag on the ground and drop the bag on top of it." ),
+                         z->get_name() );
+            }
+
+            const SpeechBubble &speech_no_time = get_speech( "mon_grocerybot_friendship_done" );
+            sounds::sound( z->pos(), speech_no_time.volume, sounds::sound_t::speech, speech_no_time.text );
+            z->remove_effect( effect_paid );
+            return true;
+        }
+    } else {
+        const time_duration time_left = z->get_effect_dur( effect_pet );
+        if( time_left < 1_minutes ) {
+            if( calendar::once_every( 20_seconds ) ) {
```

Comment:
Maybe increase this to at least once per minute?

Reply:
It only appears when there's less than 1 minute left, so it should play 3 times max.

Reply:
Oh, didn't notice that.